Scripts/script-DBotPredictPhishingEvaluation.yml

```diff
@@ -0,0 +1,168 @@
+      'F1': f1_score,
+      'Size': len(labeled_data)
+  }
+  demisto.results({
+      'ContentsFormat': formats['json'],
+      'Type': entryTypes['note'],
+      'Contents': eval_result,
+      'EntryContext': {
+          'DBotPredictPhishingEvaluation': eval_result
+      },
+      'HumanReadable': hr1 + "\n\n" + hr2,
+      'HumanReadableFormat': formats["markdown"]
+  })
+  details_string = 'F1:%.2f | Precision:%.2f | Recall:%.2f | SamplesCount: %d' % (f1_score, precision_score, recall_score, len(labeled_data))
+  demisto.executeCommand("setIncident", {'details': details_string})
+
+
+type: python
+tags:
+- ml
+- phishing
+comment: Evaluate phishing model created by text classification ml automation.
+enabled: true
+args:
+- name: incidentsQuery
+  description: Query of the phishing incidents.
+  defaultValue: created:>="3 days ago" and type:"Phishing"
+- name: maxNumberOfIncidents
+  description: Maximum number of incidents
+  defaultValue: "100"
+- name: emailTextKey
+  description: Incident key to extract email text
+  defaultValue: details
+- name: emailSubjectKey
+  description: Incident key to extract email subject
+  defaultValue: name
+- name: tagKey
+  description: Incident key to extract tag
+  defaultValue: closeReason
+- name: phishingLabels
+  description: 'Comma-separated values of email tags values and mapping. The script
+    going to consider only the tags specify in this field. You can map label to another
+    value by using this format: LABEL:MAPPED_LABEL. For example: let''s say we have
+    4 values in email tag: malicious, credentials harvesting, inner communitcation,
+    external legit email, unclassified. While training, we want to ignore "unclassified"
+    tag, and refer to "credentials harvesting" as "malicious" too. Also, we want to
+    merge "inner communitcation" and "external legit email" to one tag called "non-malicious".
+    The input will be: malicious, credentials harvesting:malicious, inner communitcation:non-malicious,
+    external legit email:non-malicious'
+  defaultValue: '*'
+- name: isContextNeeded
+  auto: PREDEFINED
+  predefined:
+  - "yes"
+  - "no"
+  description: Does one of the fields is in the context data?
+  defaultValue: "yes"
+- name: hashData
+  auto: PREDEFINED
+  predefined:
+  - "yes"
+  - "no"
+  description: Is phishing model based on hashed data?
+  defaultValue: "no"
+- name: modelListName
+  required: true
+  description: Demisto list name that stores the machine learning model
+outputs:
+- contextPath: DBotPredictPhishingEvaluation.Precision
+  description: Precision score (0-1)
+  type: number
+- contextPath: DBotPredictPhishingEvaluation.Recall
+  description: Recall score (0-1)
+  type: number
+- contextPath: DBotPredictPhishingEvaluation.F1
+  description: F1 score (0-1)
+  type: number
+- contextPath: DBotPredictPhishingEvaluation.Size
+  description: Test data size
+  type: number
+scripttarget: 0
+timeout: 6µs
+runonce: true
+dockerimage: demisto/machine-learning
```

Comment:
don't prefer to use `demisto/dl:1.0` for this?

Reply:
changed